lib/content-services/src/lib/dialogs/node-lock.dialog.ts

```diff
@@ -55,18 +53,20 @@ export class NodeLockDialogComponent implements OnInit {
         const { node } = this.data;
         this.nodeName = node.name;
 
+        const isTimeLock = !!node.properties['cm:expiryDate'];
+        const time = isTimeLock ? new Date(node.properties['cm:expiryDate']) : new Date();
```

Comment:
The one problem I can see here when we don't specify a format to work with would be how it would distinguish between dates and months when they are both less than 12. For e.g., is 04/03/2023 treated as 4th March, 2023, or 3rd April 2023.?
The older code also would probably have this bug, since no format is provided there either

Reply:
we need better typing support for JS-API, actually it comes already as a Date object from what I remember, so you don't need to format it (already includes time zones, etc)